src/pages/index.js

```diff
@@ -115,36 +128,56 @@ modals.forEach((modal) => {
   });
 });
 
-// document.addEventListener("click", (evt) => {
-//   if (!evt.target.classList.contains("modal")) {
-//     closeModal(modal);
-//   }
-// });
-
 function closeModal(modal) {
   modal.classList.remove("modal_opened");
   document.removeEventListener("keyup", handleEscape);
 }
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
+  handleLoadingState(editModalSubmit, true);
+  api.editUserInfo({name:  editModalNameInput.value, about: editModalDescriptionInput.value})
+    .then((data) =>{
+  profileName.textContent = data.name;
+  profileDescription.textContent = data.about;
   closeModal(editModal);
-}
+    })
+    .catch(console.error)
+    .finally(() =>{
+      handleLoadingState(editModalSubmit, false)}
+    )};
 
 function handleAddFormSubmit(evt) {
   evt.preventDefault();
-  const inputValues = {
-    link: addModalLinkInput.value,
-    name: addModalNameInput.value,
-  };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  addModalLinkInput.value = "";
-  addModalNameInput.value = "";
-  disableButton(addCardSubmit, settings);
-  closeModal(addCardModal);
+  handleLoadingState(addCardSubmit, true);
+  api.createCardInfo({link: addModalLinkInput.value , name:addModalNameInput.value })
+  .then((data) =>{
+    const cardElement = getCardElement(data);
+    cardsList.prepend(cardElement);
+    data.link = "";
+    data.name = "";
+    disableButton(addCardSubmit);
+    closeModal(addCardModal);
+  })
+  .finally(() =>{
+  handleLoadingState(addCardSubmit, false)}
+)};
+
+function handleAvatarFormSubmit(evt){
+  evt.preventDefault();
+  handleLoadingState(avatarCardSubmit, true);
+  api.editAvatarInfo(avatarModalInput.value)
+  .then((data) => {
+    avatarPicture.src = data.avatar;
+    closeModal(avatarModal);
+  })
+  .finally(() =>{
+  handleLoadingState(avatarCardSubmit, false)});
+}
+
+function handleLoadingState(button, isLoading) {
+  button.disabled = isLoading;
```

Comment:
` button.disabled = isLoading;`


Please, delete it because it makes problems for **disabling** the submit button in `.then` (in `.finally` it gets enabled)